### PR TITLE
Refactor Reference model to use CSL-JSON structure

### PR DESCRIPTION
**Pull request by zsarnoczay**
## Description

This pull request refactors the `Reference` model from rigid, separate fields to a flexible, standardized CSL-JSON structure. The change enables consistent citation data management using the industry-standard Citation Style Language JSON format while maintaining backward compatibility through a comprehensive 6-step migration process.

---

## Summary of Changes

This pull request implements a major refactoring of the `Reference` model and introduces a full test suite to support it. The changes were implemented via a safe, multi-step migration process.

Key changes include:
- **Model Refactoring:** The `Reference` model now uses a `csl_data` `JSONField` as the single source of truth.
- **Data Denormalization:** The `title`, `author`, and `year` fields were retained as denormalized fields that are automatically populated from `csl_data` during model save operations. These fields are marked as `editable=False` and serve as optimized query fields while maintaining `csl_data` as the authoritative source.
- **Data Migration:** A comprehensive 5-step migration process safely transforms existing data: (1) Add nullable `csl_data` field, (2) Populate `csl_data` from legacy fields using sophisticated parsing logic with CSL schema validation, (3) Remove legacy fields (`citation`, `doi`, `name`, `publication_type`) and add denormalized fields, (4) Synchronize denormalized fields using model save logic, (5) Make `csl_data` required. The migration includes advanced author parsing and publication type detection.
- **Test Suite Implementation:** A comprehensive test suite was initialized for the `ned_app`, including a new `tests/` package structure. The tests cover model logic, serializer validation, and data processing, currently limited in scope to handle the Reference model only.
- **Bug Fixes:** Enhanced data validation and error handling throughout the Reference model, including proper validation of CSL-JSON structure, robust author name parsing with support for various formats (family/given names, literal names, "et al." handling), and comprehensive edge case handling for malformed or missing data fields.

---

## How to Test

All changes are covered by the new automated test suite.

**For a fresh database setup:** Simply run `python manage.py migrate` as usual.

**For an existing database where `ned_app` tables already exist:** You must first synchronize the migration history before applying the new changes.

1.  **Synchronize the initial migration:**
    ```bash
    python manage.py migrate ned_app 0001_initial --fake
    ```
2.  **Apply all subsequent migrations:**
    ```bash
    python manage.py migrate ned_app
    ```
3.  **Run the test suite:**
    ```bash
    python manage.py test ned_app
    ```
4.  Confirm that all tests pass successfully.

### CODE REVIEW

File: ned_app/schemas/csl-data.json
[the remark this answers is not in the document]
Comment (zsarnoczay) [reply ned_app/schemas/csl-data.json:1]: I think that's a good idea. The csl_schema in there is the official one that was stored in the repo just for convenience so that we don't need to fetch it from a URL every time we need to validate an input.

File: resources/example_data/reference.json [hunk 1]
```diff
@@ -1,26 +1,43 @@
...
-        "author": "Arash E. Zaghi, E. Maragakis, Ahmad Itani, Elliot Goodwin",
-        "year": 2012,
         "study_type": "Experiment",
         "comp_type": "Hospital Piping Assembly",
-        "doi": "https://doi.org/10.1193/1.3672911",
-        "citation": "",
-        "publication_type": "Unknown",
-        "pdf_saved": "True"
+        "pdf_saved": "True",
+        "csl_data": {
```
[the remark this answers is not in the document]
Comment (zsarnoczay) [reply resources/example_data/reference.json:7]: One more thought: besides the schema, it would be helpful for contributors to have a clear documentation of expectations. We could have this in the readme file, or in a separate markdown document. It seems reasonable to me to develop this document in parallel with the review/update of the tables and metadata. Perhaps we can talk about this in our next meeting.

File: resources/example_data/reference.json [hunk 2]
```diff
@@ -1,26 +1,43 @@
 [
     {
         "id": "ARA-2012-EXP",
-        "name": "Experimental and Analytical Studies of Hospital Piping Assemblies Subjected to Seismic Loading",
-        "author": "Arash E. Zaghi, E. Maragakis, Ahmad Itani, Elliot Goodwin",
-        "year": 2012,
         "study_type": "Experiment",
```
[the remark this answers is not in the document]
Comment (zsarnoczay) [reply resources/example_data/reference.json:4]: I am leaving them for now, yes. 

I suggested removing comp_type, but for other reasons unrelated to the metadata. I thought we can discuss those later and did not want to mix that into the metadata prototyping.

Regarding study_type, I felt that it is useful to have that as a feature and the values are already quite well defined for querying. I suggested renaming it to data_type, but that's a nuance and again, I thought this can be done later, separately from the metadata prototyping.

File: resources/example_data/reference.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (dustin-cook) [reply resources/example_data/reference.json:7]: leaving the csl schema alone and clarifying our project specific requirements makes sense to me

I also agree that clearly articulating expectations to the user is a good idea, whether through an intuitive/human-readable schema file or a separate markdown file (I could see it either way). However, would the schema / readme of expectation just inherent all from the model definitions in Django? Would it be possible to just auto generate those documentation files instead of creating multiple sources of truth?
